hyperspy/tests/io/test_mrcz.py

```diff
@@ -18,134 +18,144 @@
                 try:
-                    fh = open( mrcName, 'a' )
+                    fh = open(mrcName, 'a')
                     fh.close()
                     break
                 except IOError:
                     sleep(0.001)
-            
-        reSignal = load( mrcName )
-        try: os.remove( mrcName )
-        except IOError: print( "Warning: file {} left on disk".format(mrcName) )
-
-        npt.assert_array_almost_equal( testSignal.data.shape, reSignal.data.shape )
-        npt.assert_array_almost_equal( testSignal.data, reSignal.data )
-        npt.assert_almost_equal( testSignal.metadata.get_item('Acquisition_instrument.TEM.beam_energy'), reSignal.metadata.get_item('Acquisition_instrument.TEM.beam_energy') )
-        npt.assert_almost_equal( testSignal.metadata.get_item('Acquisition_instrument.TEM.magnification'), reSignal.metadata.get_item('Acquisition_instrument.TEM.magnification') )
+
+        reSignal = load(mrcName)
+        try:
+            os.remove(mrcName)
+        except IOError:
+            print("Warning: file {} left on disk".format(mrcName))
+
+        npt.assert_array_almost_equal(
+            testSignal.data.shape,
+            reSignal.data.shape)
+        npt.assert_array_almost_equal(testSignal.data, reSignal.data)
+        npt.assert_almost_equal(
+            testSignal.metadata.Acquisition_instrument.TEM.beam_energy,
+            reSignal.metadata.Acquisition_instrument.TEM.beam_energy)
+        npt.assert_almost_equal(
+            testSignal.metadata.Acquisition_instrument.TEM.magnification,
+            reSignal.metadata.Acquisition_instrument.TEM.magnification)
         # Comparing axes is a bit difficult, due to the mixed format ordering
         # print( testSignal.axes_manager )
         # print( reSignal.axes_manager )
         # for I in range(3):
         #     print( "Test #{}  name: {}, scale: {:.3f}".format(I, testSignal.axes_manager[I].name, testSignal.axes_manager[I].scale ) )
-        #     print( "Re   #{}  name: {}, scale: {:.3f}".format(I, reSignal.axes_manager[I].name, reSignal.axes_manager[I].scale ) )
-        for aName in ['x','y','z']:
-            npt.assert_equal( testSignal.axes_manager[aName].size, reSignal.axes_manager[aName].size )
-            npt.assert_almost_equal( testSignal.axes_manager[aName].scale,  reSignal.axes_manager[aName].scale )
-            
+        # print( "Re   #{}  name: {}, scale: {:.3f}".format(I,
+        # reSignal.axes_manager[I].name, reSignal.axes_manager[I].scale ) )
+        for aName in ['x', 'y', 'z']:
+            npt.assert_equal(
+                testSignal.axes_manager[aName].size,
+                reSignal.axes_manager[aName].size)
+            npt.assert_almost_equal(
+                testSignal.axes_manager[aName].scale,
+                reSignal.axes_manager[aName].scale)
+
+        # These assertions need to be True
+#        assert testSignal.axes_manager.as_dictionary() == reSignal.axes_manager.as_dictionary()
```

Comment:
At the moment, this assertion does not pass. Since I don't a reference file, I am a bit lost to find out if the error come from the file is written or read. Also one think that is not really clear: is the z axis supposed to be the "navigation axis"?

Reply:
Officially all MRC files are C-ordered image cubes with axes [z,y,x].  This is the spec:

http://www.ccpem.ac.uk/mrc_format/mrc2014.php

If you want to maintain your Hyperspy unique axes scheme, I would suggest you bundle the `axes_manager.as_dictionary` as a field in the JSON meta-data and then pull it out again on the read.  The default behavoir if it is not present should be adhere to the MRC2014 standard (i.e. no fancy axes).

Reply:
Saving the axes_manager is not needed, however I would expect that the same `axes_manager` should be restored after writing and reading data. Also, the `as_dictionnary` method is quite convenient to check that the two signals have the same axes_manager.

Reply:
Probably the best approach would be for the file save method to add to the metadata.  E.g.:

in `file_writer`:

   meta['hyperspy_axes_manager'] = signal.axes_manager.as_dictionary()

in `file_reader`:

    if 'hyperspy_axes_manager' in meta:
        something.axes_manager.create_model( dictionary=meta['hyperspy_axes_manager'] )

It might be necessary to add an additional level of abstraction to `axes_manager` to allow it to be serialized by JSON, for example if there is non-standard Python objects in the axes_manager such as `ndarray` objects.  I.e. `to_json` wraps `as_dictionary` and `from_json` wraps `create_model`.

Reply:
How's it going with this?  Is `create_model` the correct way to build the `axes_manager` from a dict or is there an alternative way?

Reply:
It was just the order of parsed axes list, the order is not important but it makes testing easier. Now, everything is tested and it seems to work fine.